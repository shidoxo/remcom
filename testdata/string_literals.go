package main

import "fmt"

func main() {
	// This comment should be removed
	str1 := "This // is not a comment inside string"
	str2 := "URL: https://example.com" // This comment should be removed

	// More complex string cases
	url := "https://api.example.com/v1/users"   // API endpoint comment
	comment := "Use // for line comments in Go" // Explaining comment syntax

	// Escaped quotes
	escaped := "He said \"// is for comments\" to me" // Quote explanation

	// Single quotes (runes)
	char := '/'         // Single slash character
	doubleSlash := '\\' // Backslash character, this comment should be removed

	// Mixed quotes
	mixed := "Single quote: ' and // inside" // This comment removed

	fmt.Println(str1, str2, url, comment, escaped, char, doubleSlash, mixed)
}
