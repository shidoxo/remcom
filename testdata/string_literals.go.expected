package main

import "fmt"

func main() {
	str1 := "This // is not a comment inside string"
	str2 := "URL: https://example.com"

	url := "https://api.example.com/v1/users"
	comment := "Use // for line comments in Go"

	escaped := "He said \"// is for comments\" to me"

	char := '/'
	doubleSlash := '\\'

	mixed := "Single quote: ' and // inside"

	fmt.Println(str1, str2, url, comment, escaped, char, doubleSlash, mixed)
}
